### PR TITLE
bossa-arduino: init at 1.9.1-arduino2

### PR DESCRIPTION
**Pull request by ners**
###### Description of changes

Introducing Arduino's fork of the BOSSA flashing utility for flashing Arduino chips. It's what Arduino uses in its IDE and CLI utility.

I've tested in in practice in the following repository: https://github.com/ners/hello-arduino

I've also updated the vanilla BOSSA package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vojta001** [comment]
Result of `nixpkgs-review pr 186759` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>bossa</li>
    <li>bossa-arduino</li>
  </ul>
</details>

**Comment by ners** [comment]
I'm not sure how to request a review. @ryantm could you have a look?

**Comment by ners** [comment]
@SuperSandro2000 could I ask you for review?

File: pkgs/development/embedded/bossa/arduino.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{ bossa, git, fetchFromGitHub }:
+
+bossa.overrideAttrs (attrs: rec {
+  version = "1.9.1-arduino2";
+  src = fetchFromGitHub {
+    owner = "arduino";
+    repo = "BOSSA";
+    rev = version;
+    sha256 = "sha256-sBJ6QMd7cTClDnGCeOU0FT6IczEjqqRxCD7kef5GuY8=";
+    nativeBuildInputs = attrs.nativeBuildInputs ++ [ git ];
```
Comment (SuperSandro2000) [inline pkgs/development/embedded/bossa/arduino.nix:10]: That line is strange, why do we need that?

File: pkgs/development/embedded/bossa/arduino.nix [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{ bossa, git, fetchFromGitHub }:
+
+bossa.overrideAttrs (attrs: rec {
+  version = "1.9.1-arduino2";
```
Comment (SuperSandro2000) [inline pkgs/development/embedded/bossa/arduino.nix:4]:
```suggestion
  pname = "bossa-arduino";
  version = "1.9.1-arduino2";

```

**Pull request merged**